app/models/step_by_step_page.rb

```diff
@@ -177,6 +178,10 @@ def generate_content_id
     self.content_id ||= SecureRandom.uuid
   end
 
+  def strip_slug_spaces
+    self.slug.gsub!(/^\s+|\s+$/, "")
```

Comment:
As this is the same logic used by the secondary links, could we move this method to a central place to be used by both? Perhaps in `ApplicationHelper`?

Reply:
I'm not too keen on doing this, since whilst the regex is the same, what's actually happening is different.  One place is updating a value on the model, whilst the other is modifying a local variable.  It means I'd still have a `strip_slug_spaces` method in the model, which would then be referencing another method elsewhere, which feels like extra complexity.